session.go

```diff
@@ -606,24 +622,24 @@ func (s *Session) mux(remoteBegin *frames.PerformBegin) {
 				deliveryID = nextDeliveryID
 				fr.DeliveryID = &deliveryID
 				nextDeliveryID++
-				deliveryIDByHandle[fr.Handle] = deliveryID
+				deliveryIDFromOutputHandle[fr.Handle] = deliveryID
 
-				// add to handleByDeliveryID if not sender-settled
 				if !fr.Settled {
-					handlesByDeliveryID[deliveryID] = fr.Handle
```

Comment:
This really was the bug.  We mapped the delivery ID to the output (i.e. our) handle.